client/src/actions/actions.js

```diff
@@ -15,31 +16,18 @@ export const updateLink = (newLink) => {
     };
 };
 
-export const addSong = (link) => (dispatch) => {
-    dispatch({ type: BEGIN_DOWNLOAD_SONG_INFO, link });
-    const videoId = getVideoIdFromLink(link);
-    const url = formatString(config.server.resources.getVideoInfo, [videoId]);
-    fetch(url, {
-        method: 'GET',
-        headers: {
-            Accept: 'application/json'
-        },
-    })
-    .then(response => {
-        if (response.ok) {
-            response.json().then(songInfo => {
-                dispatch(downloadSongInfoSuccess(songInfo));
-            })
-        }
-        else {
-            response.json().then(jsonBody => {
-                dispatch(downloadSongInfoError(jsonBody.message));
-            })
-        }
-    })
-    .catch((e) => {
-        dispatch(downloadSongInfoError(config.download_song_info_error_message))
-    });
+export const addSong = songId => async dispatch => {
```

Comment:
BTW - that's still not an `action` (notice how it returns a method that gets `dispatch` as a prop - unlike the other methods in this file).
This is a `Thunk` action - and should be in a respective file

Reply:
so how the file structure should look like?

Reply:
Just create a new file named `thunk-actions.js`, or something of this sort, and move this method to the new file

Reply:
exactly what I did.